articles/active-directory/develop/reference-third-party-cookies-spas.md

```diff
@@ -67,7 +67,17 @@ There are two ways of accomplishing sign-in:
 
 ### A note on iframe apps
 
-A common pattern in web apps is to use an iframe to embed one app inside another. The top-level frame handles authenticating the user, and the application hosted in the iframe can trust that the user is signed in, fetching tokens silently using the implicit flow. Silent token acquisition no longer works when third-party cookies are blocked - the application embedded in the iframe must switch to using popups to access the user's session as it can't navigate to the login page.
+A common pattern in web apps is to use an iframe to embed one app inside another. The common assumption from app developers is that, top-level frame handles authenticating the user, and the application hosted in the iframe is expected to trust that the user is signed in and fetch the tokens silently. However, there are couple of caveats to this assumption irrespective of whether third-party cookies are enabled or blocked in the browser.
+
+1. Applications running in an iframe (regardless of protocol) will not be able to do to redirect-based interaction, as this is blocked by eSTS for security reasons. 
+    - The one exception is B2C applications using the Embedded Login feature: https://docs.microsoft.com/en-us/azure/active-directory-b2c/embedded-login?pivots=b2c-custom-policy 
+2. Applications running in an iframe must use silent or popup-based interactions for SSO. 
+    - Popup based interaction requires MSAL.js. Our guidance to invoke the authentication from MSAL.js via popup method can be found [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/iframe-usage.md#using-msal-in-iframed-apps). 
+    - Apart from MSAL.js library, none of the other MSAL flavors(MSAL.Java or MSAL.net etc.) do not provide a way to invoke the authetication via popup method. 
```

Comment:
I don't think this line is relevant. The rest of the doc only calls out MSAL.js v2 so we can stay focused on MSAL.js library. 
```suggestion
```